Automattic/pocket-casts-android
# Add Download episode action to shelf list in Now Playing

## Description
- This PR adds a download button to podcast shelf list. See https://github.com/Automattic/pocket-casts-android/issues/1860

Fixes #1860

## Testing Instructions
1. Play an episode
2. Go to `Now playing`
3. Click on the 3 dots at the bottom
4. Tap on `Download` button
5. ✅ Ensure the episode was queued and the download started
6. ✅ Ensure you see the notification download
7. Open the shelf item list
8. If your download did not finish yet, ensure you see the `Downloading` button
9. Once the download finished, ensure you see `Remove downloaded file`
10. Tap on `Remove downloaded file`
11. ✅ Ensure your download file was removed

## Screenshots or Screencast 
[Screen_recording_20240610_101140.webm](https://github.com/Automattic/pocket-casts-android/assets/42220351/b2333f50-ca39-4025-9280-6bd7a1fb59f0)


## Checklist
 
#### I have tested any UI changes...
